### ISSUE TITLE
multer (1.0.3) hangs single / multi file upload

### ISSUE THREAD

**Issue by siyb** [expressjs/multer#213]
version: 1.0.3

Multer sometimes hangs on file uploads, predominantly, when multiple files are uploaded, but it's also reproducible with single file uploads. I've tracked the issue to make-middleware.js, when multer hangs readFinished is false and the busboy "finish" callback is never called. I have tried uploading multiple as well as single files using request (2.61.0) and superagent (1.3.0). Strangely enough, I was not able to reproduce this issue using curl or httpie, so maybe this issue isn't entirely (or at all) related to multer or busboy. Here is my "testcode" (relevant parts):

superagent:
```javascript
 var files = ['filepath','filepath',...];
 var r = superagent
                    .post(serviceUrl)
                    .field('someField', 'value');

                _.map(files, function (i) {
                    return fs.createReadStream(i);
                }).forEach(function (file) {
                    r.attach('upload', file);
                });

                r.end(function (err, res) {
                    if (err) {
                        return reject(err);
                    } else {
                        return resolve(res);
                    }
                });
```
request:
```javascript
                var formData = {
                    upload: _.map(files, function (i) {
                        return fs.createReadStream(i);
                    }),
                    someField: 'value'
                };

                request.post({url: serviceUrl, formData: formData}, function (err, res, body) {
                    if (err) {
                        logger.error('Error uploading file', err);
                        reject(err);
                    } else {
                        resolve({res: res, body: body});
                    }
                });
```
multer:
```javascript
var uploadFile = upload.array('upload'); // using storage with standard stuff, all callbacks are called properly
```

**Comment by siyb** [comment]
might be related to #205

**Comment by DarianAnjuhal** [comment]
I had the same fight two days ago:

Try this:

```javascript
var req = request.post({url: url}, function optionalCallback(err, httpResponse, body) {

        //DO STUFF        
          });

          var form = req.form();
          form.append('file', fs.createReadStream(pathToFile));
```

Please let me know if you know more about this issue.

**Comment by siyb** [comment]
@DarianAnjuhal: thanks for the advise, I've rewritten the code to use native curl bindings now, this issue has disappeared. If I find the time, I will try it your way and report back to you ;)

**Comment by qlqllu** [comment]
I have the same issue. I use postman to post a file, the server uses multer to parse multipart form.
I can upload the file at first, but if I post some times, the server will hang and never respond.

**Comment by LinusU** [comment]
This should be fixed by #205 and released on npm as version `1.0.5` :rocket: 

Please reopen if issue persists.
